packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -1647,6 +1652,187 @@ function cleanupParameterData(inputData: NodeParameterValueType): NodeParameterV
+	>,
+):
+	| INodePropertyOptions
+	| INodeProperties
+	| INodePropertyCollection
+	| INodePropertyMode
+	| undefined {
+	const prop = options.find((i) => i.name === name);
+	return prop;
+}
+
+function findPropertyFromParameterName(
+	parameterName: string,
+	nodeType: INodeType,
+): INodePropertyOptions | INodeProperties | INodePropertyCollection {
+	let property: INodePropertyOptions | INodeProperties | INodePropertyCollection | undefined;
+	const paramParts = parameterName.split('.');
+
+	property = findPropertyInArray(paramParts.shift()!, nodeType.description.properties);
+	if (!property) {
+		throw new Error(`Couldn't not find property "${parameterName}"`);
+	}
+
+	// eslint-disable-next-line no-restricted-syntax
+	for (const p of paramParts) {
+		const param = p.split('[')[0];
+		if ('options' in property && property.options) {
+			property = findPropertyInArray(param, property.options);
+		} else if ('values' in property) {
+			property = findPropertyInArray(param, property.values);
+		} else {
+			throw new Error(`Couldn't not find property "${parameterName}"`);
+		}
+		if (!property) {
+			throw new Error(`Couldn't not find property "${parameterName}"`);
+		}
+	}
+
+	return property;
+}
+
+function executeRegexExtractValue(
+	value: string,
+	regex: RegExp,
+	parameterName: string,
+): NodeParameterValueType | object {
+	const extracted = regex.exec(value);
+	if (!extracted) {
+		throw new Error(
+			`extractValue for "${parameterName}" could not extract value. This likely means that the supplied value doesn't match the extractor regex.`,
+		);
+	}
+	if (extracted.length < 2 || extracted.length > 2) {
+		throw new Error(
+			`Property "${parameterName}" has an invalid extractValue regex "${regex.source}". extractValue expects exactly one group to be returned.`,
+		);
+	}
+	return extracted[1];
+}
+
+function extractValueRLC(
+	value: NodeParameterValueType | object,
+	property: INodeProperties,
+	parameterName: string,
+	nodeType: INodeType,
+): NodeParameterValueType | object {
+	// Not an RLC value
+	if (typeof value !== 'object' || !value || !('mode' in value) || !('value' in value)) {
+		return value;
+	}
+	const modeProp = findPropertyInArray(value.mode as string, property.modes || []);
+	if (!modeProp) {
+		return value.value;
+	}
+	if (!('extractValue' in modeProp) || !modeProp.extractValue) {
+		return value.value;
+	}
+
+	if (typeof value.value !== 'string') {
+		let typeName: string | undefined = value.value?.constructor.name;
+		if (value.value === null) {
+			typeName = 'null';
+		} else if (typeName === undefined) {
+			typeName = 'undefined';
+		}
+		throw new Error(
+			`Only strings can be passed to extractValue. Parameter "${parameterName}" passed "${typeName}"`,
+		);
+	}
+
+	if (modeProp.extractValue.type !== 'regex') {
+		throw new Error(
+			`Property "${parameterName}" has an unknown extractValue type "${
+				modeProp.extractValue.type as string
+			}"`,
+		);
+	}
```

Comment:
TS would prevent anything other than `type: 'regex'` from compiling. The assertion prolly hints that this can never happen. Why do we need this check?

Reply:
If a new type of extractor is added but the relevant hasn't been. Also, `npm run watch` doesn't always fail to compile if there's a type error, it might only warn. So this will also aid in debugging in that case.